### PR TITLE
Feature/refactor: Refactor Code For Online Training 

### PR DESCRIPTION
**Pull request by zyksir**
## Motivation

- Organize the specforge dir. Now all the code related to draft model is under `specforge/modeling/draft`, all the code related to target model is under `specforge/modeling/target`
- Organize the tests dir. Delete some useless code.
- Move the benchmark for kernel fusion to `benchmark/benchmark_module`
- Use `patch` to patch code for sglang target backend
- Fix the bug about ZeRO2 of optimization states saving. Previously we will lose some information about states on other nodes.
- Add Trainer to wrap the logic for online training.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

File: specforge/modeling/draft/base.py [hunk 1]
```diff
@@ -24,16 +24,76 @@
...
 from huggingface_hub import snapshot_download
 from safetensors import safe_open
+from transformers import PretrainedConfig
 from transformers.cache_utils import Cache
 from transformers.modeling_utils import PreTrainedModel
 
-from specforge.modeling._mask_utils import _expand_mask, _make_causal_mask
+
+@torch.no_grad()
+def load_param_from_target_model(
```
Comment (zyksir) [inline specforge/modeling/draft/base.py]: When We uniform the offline training with online training, this function can be used to load target head as well.

File: specforge/modeling/eagle3.py [hunk 2]
```diff
@@ -220,21 +233,12 @@ def __init__(
...
-        attention_mask: torch.Tensor,
-        loss_mask: torch.Tensor,
-        device: Optional[torch.device] = None,
-        **kwargs,
-    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
-        hidden_states = kwargs["hidden_states"]
-        target = kwargs["target"]
-        target = self.target_head(target)
-        return hidden_states, target, loss_mask, input_ids
+        # self.target_head = (
```
Comment (zyksir) [inline specforge/modeling/eagle3.py]: I comment this out since I find that current `prepare_hidden_states.py` will save logits instead of hidden states. logits take lots of more storage than hidden states. In the future we might want to save hidden states. So I don't delete this code

**Comment by FrankLeeeee** [comment]
I think this PR is a bit too huge as it includes 6 major changes. just wondering if it is possible to split this PR into 5/6 individual PRs?
